gitlab/v4/objects/merge_requests.py

```diff
@@ -112,6 +112,22 @@ class ProjectMergeRequest(
 ):
     _id_attr = "iid"
 
+    @property
+    def reviewer_ids(self):
+        return [reviewer["id"] for reviewer in self.reviewers]
+
+    @reviewer_ids.setter
+    def reviewer_ids(self, new_reviewer_ids):
+        new_reviewers = [{"id": id} for id in set(new_reviewer_ids)]
+        new_reviewers.extend(
+            [
+                reviewer
+                for reviewer in self.reviewers
+                if reviewer["id"] in new_reviewer_ids
+            ]
+        )
+        self.reviewers = new_reviewers
+
```

Comment:
All this _could_ be omitted and the functionality could still work, but that causes attribute lookup for `mr.reviewer_ids` to raise an `AttributeError`, unless it was specifically set, which might be kind of odd.

Reply:
Yeah, I don't see any other usage of `@property` in the `gitlab/v4/objects/*.py` files. So I'm not sure about that.

@nejch What do you think?

Reply:
Sorry for the delay here people. Thanks for this, although I had some concerns originally so it took me a while to get back to this.
1. This will be a problem for assignees as well then, right? POST/PUT take `assignee_ids`, GET returns `assignees` (edit: ah I see it in the other comment)
2. If so it's also a problem for issues, not just MRs. Again POST/PUT take `assignee_ids`, GET returns `assignees`. There may be other resources we'd need to find

So I think we're getting into the territory of GitLab API quirks kind of, where we already closed the `public_builds` (POST/PUT) vs `public_jobs` (GET) PR, https://github.com/python-gitlab/python-gitlab/pull/1439. In that case we opened an upstream issue instead. We've also recently decided not to do too much custom work around just wrapping the API, see e.g. https://github.com/python-gitlab/python-gitlab/pull/1443.

I'd like to open an upstream issue first just to see if they'd be willing to make the naming consistent though I doubt it as it's quite explicit this way for them at least, and I think they also use `*_usernames` in list filters. If we decide to do it here though I'd like it to be as generic as possible (maybe via mixins or custom attributes, if it's feasible) for all resources where this might happen. Thanks again, I'll update with the upstream issue :) Any thoughts so far on this?

Reply:
Thanks for taking the time to review @nejch  

My thoughts about the upstream change: relying on upstream changes is probably sufficient for users of GitLab.com -- they'll eventually be fixed and the need for any other solution obviated. However, for self-hosted users who may stick to earlier versions of GitLab may still yet need an interface for the API v4 capabilities of GitLab for 12.x, 13.x, etc.

I think it would be nice to see the wrapper handle this, but I would understand and support the decision to not need to maintain that code into the future.

In the meantime, perhaps we can split up the two concerns:

1. The capability to use the _assignee_ids_ for put/post
2. Whether or not we would smooth out the API quirks (e.g. with a property, or generic wrapper of some kind)

So, maybe this PR can remove the `@property` bits for now so we can proceed with adding `assignee_ids` to the list of parameters to get the basic functionality working. Then, perhaps we can discuss smoothing the API issues in a separate MR, if we still feel it's worth pursuing.

```suggestion
```

Reply:
Yes that would make sense so we don't block this for now, thanks :) and we can discuss the rest in the issue/follow-up.

Maintaining compatibility based on gitlab versions is a whole other topic IMO as gitlab keeps breaking the API behavior in subtle ways even on minor/patch releases as we saw with 13.12, so we'd potentially have to litter our code with conditionals to cover all use cases, but I understand your point, another thing to discuss :)